components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -458,7 +460,9 @@ class GeckoEngineSession(
                 interceptor != null && (!request.isDirectNavigation || interceptor.interceptsAppInitiatedRequests())
             ) {
                 val engineSession = this@GeckoEngineSession
-                val isSameDomain = engineSession.currentUrl?.tryGetHostFromUrl() == request.uri.tryGetHostFromUrl()
+                val isSameDomain = lastOnLoadRequestUrl?.tryGetHostFromUrl()?.toDisplayUrl() ==
+                    request.uri.tryGetHostFromUrl().toDisplayUrl()
+                lastOnLoadRequestUrl = request.uri
```

Comment:
Your setting this in both branches of the if statement.

Reply:
I'll simplify this. Thanks,